### PR TITLE
calcFrame10

### PR DESCRIPTION
**Pull request by zeyzen**
the function calculate the 10th frame

### CODE REVIEW

File: test/calcFrame10.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+const calcFrame10 = require('../src/client/calcFrame10');
+const assert = require('assert');
+
+describe('calcFrame10', function() {
+  it('should return the correct result of frame 10', function() {
+    assert.equal(calcFrame10(new Frame(10, 10,10)), 30); // maximum score with a thord throw  (3 strikes)
```
Comment (Pyrignos) [inline test/calcFrame10.js]:
```suggestion
    assert.equal(calcFrame10(new Frame(10, 10,10)), 30); // maximum score with a third throw  (3 strikes)
```

File: test/calcFrame10.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    assert.equal(calcFrame10(new Frame(10, 5,5)), 20); // strike on the first throw
+    assert.equal(calcFrame10(new Frame(4, 4,0)), 8); // score without third throw
+    assert.equal(calcFrame10(new Frame(8, 2,7)), 17); // score with thord throw (spare)
```
Comment (Pyrignos) [inline test/calcFrame10.js]:
```suggestion
    assert.equal(calcFrame10(new Frame(8, 2,7)), 17); // score with third throw (spare)
```

File: test/calcFrame10.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    assert.equal(calcFrame10(new Frame(0, 0,0)), 0); // score with 0 point (0 fallen keel)
+    assert.equal(calcFrame10(new Frame(0, 5,0)), 5); // second  throw score
+    assert.equal(calcFrame10(new Frame(0, 10,4)), 14); // score with wpare on the second throw
```
Comment (Pyrignos) [inline test/calcFrame10.js]:
```suggestion
    assert.equal(calcFrame10(new Frame(0, 10,4)), 14); // score with spare on the second throw  
```

File: test/calcFrame10.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  it('should return the correct result of frame 10', function() {
+    assert.equal(calcFrame10(new Frame(10, 10,10)), 30); // maximum score with a thord throw  (3 strikes)
+    assert.equal(calcFrame10(new Frame(10, 5,5)), 20); // strike on the first throw
+    assert.equal(calcFrame10(new Frame(4, 4,0)), 8); // score without third throw
+    assert.equal(calcFrame10(new Frame(8, 2,7)), 17); // score with thord throw (spare)
+    assert.equal(calcFrame10(new Frame(0, 0,0)), 0); // score with 0 point (0 fallen keel)
+    assert.equal(calcFrame10(new Frame(0, 5,0)), 5); // second  throw score
+    assert.equal(calcFrame10(new Frame(0, 10,4)), 14); // score with wpare on the second throw  
+    assert.equal(calcFrame10(new Frame(10, 10,4)), 24); // score with 2 strikes (at first)
+    assert.equal(calcFrame10(new Frame(0, 10,10)), 20); // score with 2 strike  (at last)
```
Comment (Pyrignos) [inline test/calcFrame10.js]:
```suggestion
    assert.equal(calcFrame10(new Frame(0, 10,10)), 20); // score with 2 strikes (at last)
```

File: test/calcFrame10.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  });
+});
```
Comment (Pyrignos) [inline test/calcFrame10.js]: Add tests to check if function fails
For example, ``new Frame(9, 0, 9)`` -> fail because you can't have a third throw without a strike/spare

File: src/client/calcFrame10.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
+function calcFrame10(frame){
+    var result = 0;
+    result = frame.getC1();
+
+    if(result === 10){ //mean we get a trike on the first throw
```
Comment (Pyrignos) [inline src/client/calcFrame10.js]:
```suggestion
    if(result === 10){ // we got a strike on the first throw
```

File: src/client/calcFrame10.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+        result += frame.getC2() + frame.getC3();
+        return result;
+    }
+    
+    result += frame.getC2();
+
+    if (result === 10){ //mean we get a spare
```
Comment (Pyrignos) [inline src/client/calcFrame10.js]:
```suggestion
    if (result === 10){ // we got a spare 
```

File: src/client/calcFrame10.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+        return result;
+    }
+    
+    result += frame.getC2();
+
+    if (result === 10){ //mean we get a spare 
+        result += frame.getC3();
+        return result;
+    }
+    else{ //we got a normal round with only 2 throw
```
Comment (Pyrignos) [inline src/client/calcFrame10.js]:
```suggestion
    else{ // we got a normal round with only 2 throws
```

**Review by Pyrignos — COMMENTED** [review]
Mostly subtle comments corrections
Code itself is okay for merge

File: test/test_score.js [hunk 9]
```diff
@@ -61,5 +62,18 @@ describe('calculFrame', function(){
 
   );
  }
- 
- );
\ No newline at end of file
+ );
+
+describe('calcFrame10', function() {
+  it('should return the correct result of frame 10', function() {
```
Comment (pioupiou-devs) [inline test/test_score.js]: It could be better to create a desc for each tests

 it('test_description', function() {
assert.equal(calcFrame10(10, 5, 5),20);
});

File: src/client/score.js [hunk 10]
```diff
@@ -40,4 +40,26 @@ function calculScoreTotal(frameTable){
...
+    if(result === 10){ // we got a strike on the first throw
+        result += frame.getC2() + frame.getC3();
+        return result;
+    }
+    
+    result += frame.getC2();
+
+    if (result === 10){ // we got a spare 
+        result += frame.getC3();
+        return result;
```
Comment (pioupiou-devs) [inline src/client/score.js:83]: Maybe it will be clearer to do it in one line like
return result + frame.getC3()

File: src/client/score.js [hunk 11]
```diff
@@ -40,4 +40,26 @@ function calculScoreTotal(frameTable){
...
\ No newline at end of file
+module.exports = calculScoreTotal;
+
+function calcFrame10(frame){
+    var result = 0;
+    result = frame.getC1();
+
+    if(result === 10){ // we got a strike on the first throw
+        result += frame.getC2() + frame.getC3();
+        return result;
```
Comment (pioupiou-devs) [inline src/client/score.js:70]: Maybe it will be clearer to do it in one line like
return result + frame.getC2() + frame.getC3()

File: src/client/score.js [hunk 12]
```diff
@@ -40,4 +40,26 @@ function calculScoreTotal(frameTable){
     return sum;
 }
 
-module.exports = calculScoreTotal;
\ No newline at end of file
+module.exports = calculScoreTotal;
+
+function calcFrame10(frame){
```
Comment (pioupiou-devs) [inline src/client/score.js:48]: What if the frame is null ? undefined ?
What if C1 / C2 / C3 are equal to null ? maybe you should get the 3 values first, test and add tests on null

**Review by pioupiou-devs — CHANGES_REQUESTED** [review]

File: src/client/score.js (hunk 10 above)
Comment (zeyzen) [reply src/client/score.js:83]: to apply remark of undefined or wrong input, i will not do it in 1 line

File: src/client/score.js (hunk 11 above)
Comment (zeyzen) [reply src/client/score.js:70]: to apply remark of undefined or wrong input, i will not do it in 1 line

File: src/client/score.js (hunk 12 above)
Comment (zeyzen) [reply src/client/score.js:48]: I can't check all data from the start because of  case like "waiting thrid thow" the value will be C1 / C2 / Null. But C1+ C2 can be a valid score

Comment (pioupiou-devs) [reply src/client/score.js:48]: So you must add test like if(C1!=null) before usine their value.
